src/controllers/dynakube/oneagent/oneagent_reconciler.go

```diff
@@ -157,17 +146,17 @@ func (r *Reconciler) getOneagentPods(ctx context.Context, dynakube *dynatracev1b
 	return podList.Items, listOps, err
 }
 
-func (r *Reconciler) newDaemonSetForCR(dynakube *dynatracev1beta1.DynaKube, clusterID string) (*appsv1.DaemonSet, error) {
+func (r *Reconciler) getDesiredDaemonSet(dynakube *dynatracev1beta1.DynaKube) (*appsv1.DaemonSet, error) {
 	var ds *appsv1.DaemonSet
 	var err error
 
-	switch {
-	case r.feature == daemonset.DeploymentTypeFullStack:
-		ds, err = daemonset.NewClassicFullStack(dynakube, clusterID).BuildDaemonSet()
-	case r.feature == daemonset.DeploymentTypeHostMonitoring:
-		ds, err = daemonset.NewHostMonitoring(dynakube, clusterID).BuildDaemonSet()
-	case r.feature == daemonset.DeploymentTypeCloudNative:
-		ds, err = daemonset.NewCloudNativeFullStack(dynakube, clusterID).BuildDaemonSet()
+	switch r.feature {
```

Comment:
I like the `switch` here, but shouldn't we use the matching properties from the dynakube directly here?